src/checks/requiredTypeScript.ts

```diff
@@ -59,12 +75,23 @@ async function checkPullRequest(): Promise<boolean> {
   const filter = getIgnoreFilter()
 
   const files = await github.getPullRequestFiles(pr.number)
-  const jsFiles = files.filter((f) => isForbiddenJSFile(f.filename, filter))
-  const renamedJsFiles = files.filter(
+  const filesWithContent = await Promise.all(
+    files.map((f) =>
+      (async () => {
+        const fileContent = await getFileContent(f.filename)
+        return { ...f, fileContent }
+      })()
+    )
+  )
```

Comment:
Do the files need to be fetched or are they available in the filesystem already?

Reply:
We could actually read them from the file-system 👍

Reply:
Glad to hear that - it will definitely improve the performance of the check 🙂